### PR TITLE
spirv-fuzz: Use overflow ids when inlining functions

### LINKED ISSUE #3751: spirv-fuzz: Use overflow ids in inliner
The spirv-fuzz inliner requires exactly the right number of fresh ids to be provided.  Overflow ids should be used to cater for the case during shrinking when not enough fresh ids are present.

### PR DESCRIPTION
**Pull request by afd**
Fixes #3751.

### CODE REVIEW

File: source/fuzz/transformation_inline_function.h [hunk 1]
```diff
@@ -67,8 +67,9 @@ class TransformationInlineFunction : public Transformation {
 
   // Inline |instruction_to_be_inlined| by setting its ids to the corresponding
   // ids in |result_id_map|.
-  void AdaptInlinedInstruction(opt::IRContext* ir_context,
-                               opt::Instruction* instruction) const;
+  void AdaptInlinedInstruction(
```
Comment (Vasniktel) [inline source/fuzz/transformation_inline_function.h:70]: I would probably be good to mention in the docs for `IsApplicable` that we are using overflow ids.

File: test/fuzz/transformation_inline_function_test.cpp [hunk 2]
```diff
@@ -544,8 +546,10 @@ TEST(TransformationInlineFunctionTest, ApplyToMultipleFunctions) {
                                                      {64, 76},
                                                      {65, 77},
                                                      {66, 78}});
-  ASSERT_FALSE(
-      transformation.IsApplicable(context.get(), transformation_context));
+  ASSERT_DEATH(
+      transformation.IsApplicable(context.get(), transformation_context),
+      "Bad attempt to query whether overflow ids are available.");
```
Comment (Vasniktel) [inline test/fuzz/transformation_inline_function_test.cpp:551]: Sorry for the confusion but there seem to be no assertions in the `IsApplicable`. Why do we need an `#ifndef` here?

File: source/fuzz/transformation_inline_function.h (hunk 1 above)
Comment (afd) [reply source/fuzz/transformation_inline_function.h:70]: Done.

File: test/fuzz/transformation_inline_function_test.cpp (hunk 2 above)
Comment (afd) [reply test/fuzz/transformation_inline_function_test.cpp:551]: This is because we use NullOverflowIdSource in TransformationContext.  It doesn't even let you ask if there are overflow ids:

```
  bool HasOverflowIds() const override {
    assert(false && "Bad attempt to query whether overflow ids are available.");
    return false;
  }
```

The idea is that during fuzzing sufficient ids should be provided so that this question should never be asked.

But it does mean that one needs death tests (or to pass in a different overflow id source - one that always says no but that doesn't assert) for the tests.

Comment (Vasniktel) [reply test/fuzz/transformation_inline_function_test.cpp:551]: Ok, that makes sense. Thanks.

**Review by Vasniktel — APPROVED** [review]
